pkg/broker/receiver.go

```diff
@@ -212,11 +212,32 @@ func (r *Receiver) getTrigger(ctx context.Context, ref provisioners.ChannelRefer
 
 // shouldSendMessage determines whether message 'm' should be sent based on the triggerSpec 'ts'.
 // Currently it supports exact matching on type and/or source of events.
+// TODO this should allow returning error so the errors can be surfaced to the
+// trigger
 func (r *Receiver) shouldSendMessage(ts *eventingv1alpha1.TriggerSpec, event *cloudevents.Event) bool {
-	if ts.Filter == nil || ts.Filter.SourceAndType == nil {
+	if ts.Filter == nil {
 		r.logger.Error("No filter specified")
 		return false
 	}
+
+	// TODO what should happen if multiple filter types are specified? OR? AND?
```

Comment:
I would say that only one should be allowed. We can always relax this in the future if we want.

Reply:
Updated the documentation to reflect this and removed the comment.